homeassistant/components/weatherflow/sensor.py

```diff
@@ -0,0 +1,388 @@
+"""Sensors for the smartweatherudp integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from datetime import datetime
+from enum import Enum
+
+from pyweatherflowudp.calc import Quantity
+from pyweatherflowudp.const import EVENT_RAPID_WIND
+from pyweatherflowudp.device import (
+    EVENT_OBSERVATION,
+    EVENT_STATUS_UPDATE,
+    WeatherFlowDevice,
+)
+
+from homeassistant.components.sensor import (
+    DOMAIN as SENSOR_DOMAIN,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+    DEGREE,
+    LIGHT_LUX,
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    UV_INDEX,
+    EntityCategory,
+    UnitOfElectricPotential,
+    UnitOfIrradiance,
+    UnitOfLength,
+    UnitOfPrecipitationDepth,
+    UnitOfPressure,
+    UnitOfSpeed,
+    UnitOfTemperature,
+    UnitOfVolumetricFlux,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.util.unit_system import METRIC_SYSTEM
+
+from .const import DOMAIN, LOGGER
+
+
+@dataclass
+class WeatherFlowSensorEntityDescription(SensorEntityDescription):
+    """Describes a WeatherFlow sensor entity description."""
+
+    event_subscriptions: list[str] = field(default_factory=lambda: [EVENT_OBSERVATION])
+    imperial_suggested_unit: None | str = None
+
+
+@dataclass
+class AirDensityWeatherFlowSensorEntityDescription(WeatherFlowSensorEntityDescription):
+    """Custom class to handle the conversion between backing lib and Home Assistant Compatible VOC sensor."""
+
+    imperial_unit_of_measurement: str | None = None
+
+
+@dataclass
+class WeatherFlowWindSensorEntityDescription(WeatherFlowSensorEntityDescription):
+    """Describes a WeatherFlow wind sensor entity description."""
+
+    def __post_init__(self) -> None:
+        """Post initialisation processing."""
+        self.icon = "mdi:weather-windy"
+        self.native_unit_of_measurement = UnitOfSpeed.KILOMETERS_PER_HOUR
+        self.state_class = SensorStateClass.MEASUREMENT
+        self.suggested_display_precision = 2
+
+
+CUSTOM_SENSORS: tuple[AirDensityWeatherFlowSensorEntityDescription, ...] = (
```

Comment:
And why is it called custom

Reply:
Because it has a custom native-value option...

Reply:
I'm naming it VOC_SENSORS